### ISSUE TITLE
Pre-submission enquiry

### ISSUE THREAD

**Issue by thartbm** [openjournals/jose#52]
Hello!

I'm part of Neuromatch Academy, a large online summerschool on computational neuroscience that ran for the first time last summer. For NMA, we created 15 days of material, centered around Jupyter notebooks (running on Colab) and providing guidance through video lectures and a TA, with student and TA versions of the notebooks. Each day of material was built by a separate team of content creators. Right now we are looking for options to publish this material, so we'd like to know if you'd be interested in 15 submissions to JOSE.

All the material can be found here: https://github.com/NeuromatchAcademy/course-content

We think publishing the material in JOSE would add value for both the creators and users of this content. Creators would get recognized for their particular contributions (as they are not the ones putting the material on GitHub). Users would benefit as publication in JOSE would ensure a level of quality and it would ensure that others have the right to build on the material or integrate it in their own teaching or summerschool.

Best,
Marius

**Comment by moorepants** [comment]
Marius,

It seems appropriate from the face of it. What do you mean by "15 submissions"?

**Comment by kyleniemeyer** [comment]
@thartbm agree that this seems in scope, but you would submit the entire course/summer school as a single submission.

**Comment by kyleniemeyer** [comment]
(unless the content/topics are substantially different, I think)

**Comment by thartbm** [comment]
Dear Profs. Moore and Niemeyer,

Thanks for looking at our pre-submission inquiry and agreeing that this material falls in scope of JOSE.

We're still compiling and checking a list of all contributors ([here](https://docs.google.com/spreadsheets/d/1K4FPRrj-iSadPN_dXIM3Fv-zPXe8SvHO18dEAGYCIwU/edit#gid=1882776245)), but there are well over 100 authors. There are lecturers, notebook writers, and mini-lecturers, but also notebook code and content reviewers/editors (waxers), copyright checkers, video editors, captioners and caption translators (english, spanish and chinese captioning is available for almost all videos) as well as day-team coordinators.

We think that a single submission to JOSE is not suitable for this material, for the following reasons.

- the material is extensive: each day provides a 6 hour program, plus bonus material, with two lectures, two sets of coding exercises in 2 to 7 python/jupyter/colab notebooks and TA-led discussions
- the material is modular: each day is designed to stand on its own, and while we have taken care in later days to build on material from previous days, this only applies to broad concepts
- content creation teams, with numerous contributors (see above), operated independently within a mandate, so that crediting members of one team for work done by others seems inappropriate

We would like to provide recognition that each day forms a cohesive and independent module, so that others can sample a subset of the material to create a new summer school and credit exactly that source material they used.

We'd like to hear your thoughts on this,

NMA content creators

**Comment by moorepants** [comment]
@thartbm Thanks for the notes. Some thoughts.

We should be able to handle any number of authors. Note that this is what the reviewers check for:

>  Authorship: Has the submitting author made visible contributions to the module? Does the full list of authors seem appropriate and complete?

> the material is extensive: each day provides a 6 hour program, plus bonus material, with two lectures, two sets of coding exercises in 2 to 7 python/jupyter/colab notebooks and TA-led discussions

Note that we take single submissions for entire (multi)-semester courses, so it isn't clear why this set is any more extensive that a semester course. 

> the material is modular: each day is designed to stand on its own, and while we have taken care in later days to build on material from previous days, this only applies to broad concepts

If the materials are modular and independent of each other, with different sets of authors, then that may make sense to submit individually. Maybe you think of this like a chapter in and edited book.

If you have a single repository for each independent module, that would indicate that each chunk is likely independent and we'll see the authors associated with each repo You could also list here the titles of each independent piece and # of authors for each piece, so we could get a sense of the submissions you hope to make.

**Comment by thartbm** [comment]
Dear Profs. Moore and Niemeyer,

The material is meant to be highly modular, with specified pre-requisites that may be met elsewhere, so that others may pick and choose to build their own summerschool. A book with chapters would fit that idea.

However, a majority of people is OK with one submission to JOSE as well.

Best regards,

NMA content creators

**Comment by thartbm** [comment]
After a long silence we've sorted ourselves: who's author and who did what. We want to go with 1 paper, as this will be easier for everyone in the end. It is here: 
https://github.com/NeuromatchAcademy/course-content/tree/jose_paper

But when we test it on the (pre)submission page it throws an error:

> Your paper failed to compile with the following errors:
>  Looks like we failed to compile the PDF with the following error: Error producing PDF. ! Dimension too large. \height ->\ht \@tempboxa l.566 \end{minipage}

I can imagine this being caused by either the looong author list and affiliation list, or maybe because of one of the oversized tables at the end. However, other than the error message there is very little to go on to figure out what the cause is, let alone how to fix this. Can you advise us on how to figure this out or how to fix this?

**Comment by thartbm** [comment]
We have another question. In the markdown example the author names are just one single string, without delineation of where a last/family name starts and first/given names end (and I've noticed that I didn't follow that rule, but put last/family name first with first/given names after, separated by a comma). I've recently been added to a database as `Hart, BM'`, whereas it should have been `'t Hart, BM` but the `'t` was seen as a first/given name (and not part of my last/family name) and the first character used as an initial. There are other author names with similarly potentially unconventional formatting. Is there some way we can solve this pre-emptively?

**Comment by moorepants** [comment]
I tried to compile here https://whedon.theoj.org/ and it worked:

http://res.cloudinary.com/hju22ue2k/image/upload/v1613377775/ldfjesxp6f9nx2b6a301.pdf

The formatting of the authors is awkward, so we can ask the joss/jose maintainers about how to deal with long author lists.

We'll also have to ask about the author parsing too. Can you open a PR with the submission and then we can tag in some people to help check out the issues?

**Comment by thartbm** [comment]
Yes, I did mess up the YAML formatting, so after that was fixed it compiled: my bad. I'm still getting some small requests for changes in the author list. Might as well have the list final before getting anybody else involved, but that means I have to do some more bookkeeping on my end first, FYI (it won't be months this time).

**Comment by thartbm** [comment]
I'm a little confused about what we should be merging, as getting JOSS/JOSE maintainers involved suggests there is a JOSE clone of the repo somewhere, but I can't find that. Please advise?

**Comment by moorepants** [comment]
I'm not sure what you mean by "merging". If there are compile errors that we don't understand we can ask other people involved in JOSS and JOSE (they use the same systems). You should submit your paper via the jose website, which will trigger a pre-review issue.

**Comment by thartbm** [comment]
I have submitted the paper, and there is a status tracker page that says it has been submitted but review hasn't started. There are no compile errors anymore, but there may still be other issues (as you said, the author list is formatted strangely). I'm assuming we now wait until review has started, right?

**Comment by moorepants** [comment]
The review will show up here: https://github.com/openjournals/jose-reviews/issues and we'll work out all the details of formatting etc. there.

**Comment by thartbm** [comment]
I think we've successfully managed that step, so if you'd want to tag in people, it could be done here: https://github.com/openjournals/jose-reviews/issues/110
